core/src/test/scala/org/apache/spark/FailureSuite.scala

```diff
@@ -139,5 +139,32 @@ class FailureSuite extends SparkFunSuite with LocalSparkContext {
     FailureSuiteState.clear()
   }
 
+  test("failure cause is sent back to driver") {
+    sc = new SparkContext("local", "test")
+    val data = sc.makeRDD(1 to 3).map(x => { throw new IllegalStateException("oops"); (x,
+      x) }).groupByKey(3)
+    val thrown = intercept[SparkException] {
+      data.collect()
+    }
+    assert(thrown.getClass === classOf[SparkException])
+    assert(thrown.getCause.getClass === classOf[IllegalStateException])
+  }
```

Comment:
super minor, but can you also check the contents of the `IllegalStateException`?  I'm sure its fine in your implementation, but might as well add a check that we don't regress and somehow don't serialize the full exception